test/Motor.Extensions.Hosting.RabbitMQ_IntegrationTest/RabbitMQTests.cs

```diff
@@ -208,21 +211,98 @@ public async Task ConsumerStopAsync_ConsumeCallbackShouldStopAsEarlyAsPossible_N
         {
             var builder = RabbitMQTestBuilder
                 .CreateWithQueueDeclare(_fixture)
-                .WithConsumerCallback(async (context, bytes) =>
+                .WithConsumerCallback(async (_, token) =>
                 {
-                    await Task.Delay(4000);
+                    await Task.Delay(4000, token);
                     return ProcessedMessageStatus.CriticalFailure;
                 })
                 .WithMultipleRandomPublishedMessage()
                 .Build();
+
             var consumer = builder.GetConsumer<string>();
-            await consumer.StartAsync();
 
+            await consumer.StartAsync();
             await consumer.StopAsync();
 
             await Task.Delay(TimeSpan.FromSeconds(3));
             Assert.Equal(RabbitMQTestBuilder.PrefetchCount, builder.MessageInConsumerQueue());
         }
 
+        [Fact]
+        public async Task QueueMonitor_GetCurrentState_QueueEmpty_EmptyResult()
+        {
+            var builder = RabbitMQTestBuilder
+                .CreateWithQueueDeclare(_fixture)
+                .Build();
+
+            var monitor = new RabbitMQQueueMonitor<string>(
+                Mock.Of<ILogger<RabbitMQQueueMonitor<string>>>(),
+                Opts.Create(new RabbitMQConsumerOptions<string>
+                {
+                    Queue =
+                    {
+                        Name = builder.QueueName
+                    }
+                }),
+                _fixture.ConnectionFactory<string>()
+            );
+
+            var state = await monitor.GetCurrentState();
+            Assert.Equal(builder.QueueName, state.QueueName);
+            Assert.Equal(0, state.ConsumerCount);
+            Assert.InRange(state.ReadyMessages, 0, Int64.MaxValue);
+        }
+
+        [Fact]
+        public async Task QueueMonitor_GetCurrentState_SingleMessage_CorrespondingResult()
+        {
+            var builder = RabbitMQTestBuilder
+                .CreateWithQueueDeclare(_fixture)
+                .WithSingleRandomPublishedMessage()
+                .Build();
+
+            var monitor = new RabbitMQQueueMonitor<string>(
+                Mock.Of<ILogger<RabbitMQQueueMonitor<string>>>(),
+                Opts.Create(new RabbitMQConsumerOptions<string>
+                {
+                    Queue =
+                    {
+                        Name = builder.QueueName
+                    }
+                }),
+                _fixture.ConnectionFactory<string>()
+            );
+
+            var state = await monitor.GetCurrentState();
+            Assert.Equal(builder.QueueName, state.QueueName);
+            Assert.Equal(0, state.ConsumerCount);
+            Assert.Equal(1, state.ReadyMessages);
+        }
+
+        [Fact]
+        public async Task QueueMonitor_GetCurrentState_MultipleMessages_CorrespondingResult()
```

Comment:
Can you add tests for the `ConsumerCount` as well? All of these three tests only cover the condition where there are no consumers.

Reply:
Can do.

Reply:
see updated tests